debian/upgrade-utils/desktop-grid/meta-set.1
.\" Man Page for meta-set
.TH "meta-set" "1"
.SH "NAME"
meta-set \- utils for get file meta info.
.SH "SYNOPSIS"
.B meta-set [OPTION] <path> [keys..]
.SH "DESCRIPTION"
meta-set
This program provides several commands for set file meta info.
.TP
\fB -t, --tree\fR
Tree
.TP
\fB -u, --unset\fR
Unset
.TP
\fB -l, --list\fR
Set as list
.SH "BUGS"
.SS Should you encounter any bugs, they may be reported at: 
https://gitee.com/openkylin/peony/issues
.SH "AUTHORS"
.SS This Manual Page has been written for the UKUI Desktop Environment by:
Yue Lan <lanyue@kylinos.cn> (2024)


debian/changelog
peony (4.0.0.1-1) unstable; urgency=medium

  * Disable the auto partition mount to fix UbuntuKylin install error.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Wed, 10 Apr 2024 18:43:49 +0800

peony (4.0.0.0-1.1) unstable; urgency=medium

  * Non-maintainer upload.
  * Rename libraries for 64-bit time_t transition.  Closes: #1063217

 -- Benjamin Drung <bdrung@debian.org>  Thu, 29 Feb 2024 20:19:01 +0000

peony (4.0.0.0-1) unstable; urgency=medium

  * New upstream release.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Tue, 23 Jan 2024 10:42:22 +0800

peony (3.2.4-1) unstable; urgency=medium

  * New upstream release.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Tue, 22 Feb 2022 16:11:28 +0800

peony (3.2.3-1) unstable; urgency=medium

  * New upstream release.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Tue, 17 Aug 2021 19:35:54 +0800

peony (3.0.6-1) unstable; urgency=medium

  * New upstream release.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Sat, 28 Nov 2020 15:53:31 +0800


peony (3.0.5-1) unstable; urgency=medium

  * New upstream bugfix release.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Sun, 27 Sep 2020 19:28:06 +0800

peony (3.0.4-1) unstable; urgency=medium

  * New upstream bugfix release.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Sat, 26 Sep 2020 20:26:06 +0800

peony (3.0.3-1) unstable; urgency=medium

  * New upstream bugfix release.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Mon, 07 Sep 2020 16:16:47 +0800

peony (3.0.2-2) unstable; urgency=medium

  * debian/control:
    - libpeony3: add breaks with libpeony2. (Closes: #968595)

 -- handsome_feng <jianfengli@ubuntukylin.com>  Tue, 18 Aug 2020 13:11:28 +0000

peony (3.0.2-1) unstable; urgency=medium

  * New upstream release.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Wed, 12 Aug 2020 09:22:02 +0000

peony (3.0.1-1) unstable; urgency=medium

  * New upstream release. (Closes: #957676)
  * debian:
    - drop libgtk2.0-dev and qt5-gtk2-platformtheme.
    - update the install files.
    - libpeony2 -> libpeony3.
    - Update watch file.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Sat, 08 Aug 2020 14:22:20 +0800

peony (2.1.2-1) unstable; urgency=medium

  * New upstream bugfix release.
    - Fix enter key as shortcut issues such as open selected file.
    - Fix that file in drag state not show file name.
    - Fix that file rename as empty and rename file change current sort.
    - Fix that desktop app link show incorrect icon.
    - Fix generate file thumbnail issue and ensure thread safety.
    - Fix that screen mirror not show view.
    - Fix file handle error.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Mon, 06 Apr 2020 19:57:13 +0800

peony (2.1.1-1) unstable; urgency=medium

  * New upstream bugfix release.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Mon, 30 Mar 2020 21:42:00 +0800

peony (2.1.0-1) unstable; urgency=medium

  * New upstream release.
    - Use custom header bar, and border rounded window.
    - Remove current advance search page, and relpace it with file label
      page.
    - Improve the thumbnail frameworks.
    - Improve the mouse right click interaction in desktop view.
    - Allow unrelated trash or delete file operation executed when a
      file operation is running.
    - Refactor the desktop window background displaying process.
  * debian/libpeony3.symbols:
    - update symbols file.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Mon, 23 Mar 2020 17:56:24 +0800

peony (2.0.1-2) unstable; urgency=medium

  * debian/libpeony3.symbols:
    - fixed symbols for armel.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Thu, 20 Feb 2020 11:00:36 +0800

peony (2.0.1-1) unstable; urgency=medium

  * New upstream release.
  * debian/control:
    - Add build-depends pkg-kde-tools.
  * debian/rule:
    - using pkgkde_symbolshelper.
  * debian/libpeony3.symbols.
    - Using pkg-kde-tools update the symbols file. (Closes: #951607)

 -- handsome_feng <jianfengli@ubuntukylin.com>  Wed, 19 Feb 2020 14:15:38 +0800

peony (2.0.0-1) unstable; urgency=medium

  * New upstream release.
    - Refactor peony ui with Qt (from Gtk3).
  * debian/control:
    - Bump to compat 12.
    - Bump Standards-Version to 4.5.0.
    - Refresh the dependency.
    - Set Rules-requires-Root to no.
  * debian/copyright: Refreash the copyright entry.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Tue, 04 Feb 2020 11:09:14 +0800

peony (1.1.7-1) unstable; urgency=medium

  * Fix that the outline of the file name label is incomplete.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Wed, 16 Oct 2019 16:44:58 +0800

peony (1.1.6-1) unstable; urgency=medium

  * Add file path to the address bar that can click.
  * Add drag directories to favorites and delete favorites directories
    function.
  * Add folder image and set location button size.
  * Add search entry delete function.
  * Modify sidebar, statusbar, locationbar style.
  * Display picture according to its own width and height.
  * Update zh_TW.po and zh_CN.po.
  * Fix trash restore files error.
  * Fix that computer, recent and trash is English in addressbar.
  * Fix the location bar error.
  * Reload window after desktop-is-home-dir changed. (LP: #1831857)
  * Fix the crash when using listview mode. (LP: #1827994)
  * debian/control:
    - Drop engrampa from suggests.
    - Fix symbols-file-missing-build-depends-package-field.
    - Bump Standards-Version to 4.4.0.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Tue, 06 Aug 2019 14:51:41 +0800

peony (1.1.5-1) unstable; urgency=medium

  * Add the function of preview files.
  * Add format function.
  * Change the name style when copy files.
  * Simplify disk label description.
  * Update es.po, fr.po, pt.po, ru.po.

 -- handsome_feng <jianfengli@ubuntukylin.com>  Thu, 27 Dec 2018 20:43:29 +0800

peony (1.1.4-1) unstable; urgency=medium

  * New upstream release

 -- Aron Xu <aron@debian.org>  Mon, 22 Oct 2018 23:41:44 +0800

peony (1.1.3-1) unstable; urgency=medium

debian/upgrade-utils/desktop-grid/meta-set.c
#include "metatree.h"
#include <glib/gstdio.h>
#include <gio/gio.h>

static gboolean unset = FALSE;
static gboolean list = FALSE;
static gboolean use_dbus = FALSE;
static char *treename = NULL;
static GOptionEntry entries[] =
{
  { "tree", 't', 0, G_OPTION_ARG_STRING, &treename, "Tree", NULL},
  { "unset", 'u', 0, G_OPTION_ARG_NONE, &unset, "Unset", NULL },
  { "list", 'l', 0, G_OPTION_ARG_NONE, &list, "Set as list", NULL },
  { NULL }
};

int
main (int argc,
      char *argv[])
{
  MetaTree *tree;
  GError *error = NULL;
  GOptionContext *context;
  MetaLookupCache *lookup;
  struct stat statbuf;
  const char *path, *key;
  const char *metatreefile;
  char *tree_path;
  
  context = g_option_context_new ("<path> <key> <value> - set metadata");
  g_option_context_add_main_entries (context, entries, NULL);
  if (!g_option_context_parse (context, &argc, &argv, &error))
    {
      g_printerr ("option parsing failed: %s\n", error->message);
      return 1;
    }

  if (argc < 2)
    {
      g_printerr ("no path specified\n");
      return 1;
    }
  path = argv[1];

  if (argc < 3)
    {
      g_printerr ("no key specified\n");
      return 1;
    }
  key = argv[2];

  if (!list && !unset && argc != 4)
    {
      g_print ("No value specified\n");
      return 1;
    }

  if (treename)
    {
      tree = meta_tree_lookup_by_name (treename, TRUE);
      if (tree)
	tree_path = g_strdup (path);

      if (tree == NULL)
	{
	  g_printerr ("can't open metadata tree %s\n", path);
	  return 1;
	}
    }
  else
    {
      lookup = meta_lookup_cache_new ();
      if (g_lstat (path, &statbuf) != 0)
	{
	  g_printerr ("can't find file %s\n", path);
	  return 1;
	}
      tree = meta_lookup_cache_lookup_path (lookup,
					    path,
					    statbuf.st_dev,
					    TRUE,
					    &tree_path);
      meta_lookup_cache_free (lookup);

      if (tree == NULL)
	{
	  g_printerr ("can't open metadata tree for file %s\n", path);
	  return 1;
	}
    }

  if (unset)
    {
	{
	  if (!meta_tree_unset (tree, tree_path, key))
	    {
	      g_printerr ("Unable to unset key\n");
	      return 1;
	    }
	}
    }
  else if (list)
    {
	{
	  if (!meta_tree_set_stringv (tree, tree_path, key, &argv[3]))
	    {
	      g_printerr ("Unable to set key\n");
	      return 1;
	    }
	}
    }
  else
    {
	{
	  if (!meta_tree_set_string (tree, tree_path, key, argv[3]))
	    {
	      g_printerr ("Unable to set key\n");
	      return 1;
	    }
	}
    }

  return 0;
}
